status-reporter/status_reporter/status_reporter.py

```diff
@@ -0,0 +1,139 @@
+
+logger = logging.getLogger(STATUS_REPORTER)
+logger.setLevel(logging.INFO)
+formatter = logging.Formatter(fmt='%(asctime)s [%(levelname)s] '
+                                  '[%(name)s] %(message)s',
+                              datefmt='%d/%m/%Y %H:%M:%S')
+file_handler = WatchedFileHandler(filename=LOGFILE)
+file_handler.setFormatter(formatter)
+logger.addHandler(file_handler)
+
+
+class InitializationError(Exception):
+    pass
+
+
+class Reporter(object):
+    def __init__(self, sampler, component_type):
+        if not callable(sampler):
+            raise InitializationError('Reporter expected a callable sampler,'
+                                      ' got {0!r}..')
+        self.status_sampler = sampler
+
+        try:
+            with open(CONFIGURATION_PATH) as f:
+                self._config = yaml.safe_load(f)
+        except yaml.YAMLError as e:
+            raise InitializationError('Failed loading status reporter\'s '
+                                      'configuration with the following: {0}'.format(e))
+
+        self._reporter_user_name = self._config.get('user_name', None)
+        self._reporter_token = self._config.get('token', None)
+        self._ca_path = self._config.get('ca_path', None)
+        self._managers_ips = self._config.get('managers_ips', [])
+
+        if not all([self._managers_ips,
+                    self._reporter_token,
+                    self._reporter_user_name
+                    ]):
+            invalid_conf_settings = {
+                'Managers Ips': self._managers_ips,
+                'Username': self._reporter_user_name,
+                'Password token': self._reporter_token
+            }
+            raise InitializationError('Please verify the reporter\'s config related to '
+                                      '{0} ..'.format(json.dumps(invalid_conf_settings,
+                                                                 indent=1)))
+
+        if self._ca_path and os.path.exists(self._ca_path):
+            raise InitializationError('CA certificate was not found, '
+                                      'please verify the given location {0}'.format(self._ca_path))
+
+        self._current_reporting_freq = self._config.get('reporting_freq', None)
+        self._component_type = component_type
+
+    def _build_auth_headers(self):
+        return {
+            'Authorization': 'Basic ' + base64.b64encode('{0}:{1}'.format(
+                self._reporter_user_name,
+                self._reporter_token)
+            ),
+            'tenant': 'default_tenant'
+        }
+
+    def _build_full_reporting_url(self, ip):
+        # TODO: UUID still needs to be added here
+        return 'https://{0}/cluster_status/{1}'.format(ip, self._component_type)
+
+    def _build_report(self, status):
+        if 'reporting_freq' in status:
+            logger.error('The status report contained reserved keyword "reporting_freq",'
```

Comment:
This sounds like an error in the design of the report structure. What is this structure?